src/db/jobs.rs

```diff
@@ -0,0 +1,104 @@
+//! The `jobs` table provides a way to have scheduled jobs
+use anyhow::{Result, Context as _};
+use chrono::{DateTime, FixedOffset};
+use tokio_postgres::{Client as DbClient};
+use uuid::Uuid;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Job {
+    pub id: Uuid,
+    pub name: String,
+    pub expected_time: DateTime<FixedOffset>,
+    pub metadata: serde_json::Value,
+    pub executed_at: Option<DateTime<FixedOffset>>,
+    pub error_message: Option<String>,
+}
+
+pub async fn insert_job(db: &DbClient, job: &Job) -> Result<()> {
+    tracing::trace!("insert_job(id={})", job.id);
+    
+    db.execute(
+        "INSERT INTO jobs (id, name, expected_time, metadata, executed_at, error_message) VALUES ($1, $2, $3, $4, $5, $6) ON CONFLICT DO NOTHING",
+        &[&job.id, &job.name, &job.expected_time, &job.metadata, &job.executed_at, &job.error_message],
```

Comment:
Can we also add a unique index on (name, expected_time)? The `id` is useful for tracking + deleting jobs, but I think we'll want to avoid duplicating the same event at the same time.

Reply:
Agree!!! 💯